### DISCUSSION TITLE
Rough Edges

### DISCUSSION THREAD

**Discussion by dmrvb** [twinbasic/twinbasic#294] [General]
TwinBasic currently has a lot of rough edges, unexpected behaviours and "bugs I can live with".
At this stage in the development this is fine and to be expected, but once I start paying £300+ each year I would expect more solid code. More importantly this might give a very poor first impression for potential new users just at a time when positive publicity is required.

I just loaded a VB class into TB and its given over 1000 errors (problems). If I was a potential new user I would likely have lost interest at this stage. Its ok for me because I already have confidence that these things can be sorted, but that's only because TB has already successfully compiled quite a bit of my code.

How shall we handle these not critical bugs? Wayne really needs to be putting his time into the GUI at present and flooding the git with loads of bug reports might not be the way to go.  Or should we just report everything????

**Comment by WaynePhillipsEA** [comment]
Please report them!  Either individually, or just uploading the .twinproj file with a screenshot of the problems window would suffice.  I bet those 1000+ problems come down to only a handful of actual bugs, because we've now got a lot of real world (and highly complex) code covered in unit tests.

Compilation bugs are the highest priority above everything else.  Most are easy to fix and you'll see them fixed in a day or two after reporting.

A lot of effort will be put into improving the user experience over the next few months before the first proper release, and that includes fixing a lot of the rough edges and improving the compiler error information reported via VS code, amongst other things.

**Reply by dmrvb** [reply]
I also suspect its only a few bugs giving rise to lots of "knock on" problem reports. That's ok with me as I have some intuition as to which are the real ones that I need to fix, but maybe a bit concerning for a potential new user. I do think a good first impression will be important in the early full release of TB.
I realise that some "knock on" bugs are unavoidable but suspect some could be avoided. I will investigate and report, and send you the project if I can't work out what is going on.
